From 102ba23adbb003e4a5c626fb346423dbe53c55fe Mon Sep 17 00:00:00 2001
From: Conor Dooley <conor.dooley@microchip.com>
Date: Thu, 5 Aug 2021 10:35:27 +0100
Subject: [PATCH 57/93] riscv: icicle kit: enable signature and cert services

---
 .../microchip-mpfs-icicle-kit-context-a.dts          | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit-context-a.dts b/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit-context-a.dts
index 685d85543ca3..a2a1d9eb093c 100644
--- a/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit-context-a.dts
+++ b/arch/riscv/boot/dts/microchip/microchip-mpfs-icicle-kit-context-a.dts
@@ -511,5 +511,17 @@ fpgadigest: fpgadigest {
 			#size-cells = <1>;
 			syscontroller = <&syscontroller>;
 		};
+		devicecert: cert {
+			compatible = "microchip,polarfire-soc-device-cert";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			syscontroller = <&syscontroller>;
+		};
+		signature: signature {
+			compatible = "microchip,polarfire-soc-signature";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			syscontroller = <&syscontroller>;
+		};
 	};
 };
\ No newline at end of file
-- 
2.30.2

